crates/sui-core/src/authority_aggregator.rs

```diff
@@ -858,6 +948,167 @@ where
+                                    *total_stake += weight;
+                                    if *total_stake >= threshold {
+                                        state.committee_info = Some(CommitteeInfo {
+                                            epoch: resp.epoch,
+                                            committee_info: info,
+                                        });
+                                        return Ok(ReduceOutput::End(state));
+                                    }
+                                } else {
+                                    // This is technically unreachable because SafeClient
+                                    // does the sanity check in `verify_committee_info_response`
+                                    state.bad_weight += weight;
+                                    state.errors.push((
+                                        name,
+                                        SuiError::from("Validator returns empty committee info."),
+                                    ));
+                                }
+                            }
+                            Err(err) => {
+                                state.bad_weight += weight;
+                                state.errors.push((name, err));
+                            }
+                        };
+
+                        // Return all errors if a quorum is not possible.
+                        if state.bad_weight > validity {
+                            return Err(SuiError::TooManyIncorrectAuthorities {
+                                errors: state.errors,
+                                action: "get_committee_info",
+                            });
+                        }
+                        Ok(ReduceOutput::Continue(state))
+                    })
+                },
+                // A long timeout before we hear back from a quorum
+                self.timeouts.pre_quorum_timeout,
+            )
+            .await?;
+
+        if let Some(committee_info) = final_state.committee_info {
+            Ok(committee_info)
+        } else {
+            Err(SuiError::TooManyIncorrectAuthorities {
+                errors: final_state.errors,
+                action: "get_committee_info",
+            })
+        }
+    }
+
+    /// Query validators for latest SuiSystemState and try to form
+    /// CommitteeWithNetworkAddress. Only return Some(CommitteeWithNetAddresses)
+    /// when there is quorum. This function tolerates uninteresting
+    /// differences in SuiSystemState as long as they all link to the same
+    /// CommitteeWithNetAddresses.
+    /// This function ignores SuiSystemState that has older epoch than
+    /// `minimal_epoch`.
+    /// Usually, the caller should pass in the local incumbent epoch id
+    /// as `minimal_epoch`, to avoid getting confused by byzantine
+    /// validators.
+    pub async fn get_committee_with_net_addresses(
```

Comment:
This is ok for now, but I think we could consider simplify things by also including network information in the CommitteeStore so that other entities can query it directly without going through the system state object

Reply:
agree